benchmarks/modules/utils.py

```diff
@@ -267,11 +271,38 @@ def setup_spack_environment(self):
             f'(cd {cp_dir}; find . \( -name "spack.yaml" -o -name "compilers.yaml" -o -name "packages.yaml" \) -print0 | xargs -0 tar cf - | tar -C {dest} -xvf -)',
             f'spack -e {self.build_system.environment} config add "config:install_tree:root:{env_dir}/opt"',
         ]
-
+        # else f"{d[key].value}" if isinstance(d[key].value, bool) for boolean with "True" instead of True
+        cmd_spack_variants = 'from spack import environment; d = environment.active_environment().spec_lists["specs"].specs[0].variants.dict; keys = list(d.keys());values = {key: ([d[key].value[i] for i in range(len(d[key].value))] if isinstance(d[key].value, tuple) else d[key].value) for key in keys};print(values)'
+        cmd_spec_mpi =  'from spack import environment;print(environment.active_environment().spec_lists["specs"].specs[0]["mpi"]) if "mpi" in environment.active_environment().spec_lists["specs"].specs[0] else "" '
+        cmd_compiler_name = 'from spack import environment; print(environment.active_environment().spec_lists["specs"].specs[0].compiler.name)'
+        # Although we could provide several versions of compiler, reframe only executes with the first version
+        cmd_compiler_version = 'from spack import environment;environment.active_environment().spec_lists["specs"].specs[0].compiler.versions[0]'
+        self.postrun_cmds.append(f'echo "compiler_name: $(spack -e {self.build_system.environment} python -c \'{cmd_compiler_name}\')"')
+        self.postrun_cmds.append(f'echo "compiler_version: $(spack -e {self.build_system.environment} python -c \'{cmd_compiler_version}\')"')
+        self.postrun_cmds.append(f'echo "spack_spec_variants: $(spack -e {self.build_system.environment} python -c \'{cmd_spack_variants}\')"')
+        self.postrun_cmds.append(f'echo "MPI: $(spack -e {self.build_system.environment} python -c \'{cmd_spec_mpi}\')"')
+        
         # Keep the `spack.lock` file in the output directory so that the Spack
         # environment can be faithfully reproduced later.
         self.keep_files.append(os.path.realpath(os.path.join(self.build_system.environment, 'spack.lock')))
 
+    @run_after('run')
+    def get_compiler_name(self):
+        with osext.change_dir(self.stagedir):
+            self.compiler_name = sn.extractsingle(r'compiler_name: \s*(\S+)', self.stdout, 1).evaluate()
```

Comment:
Do we need to do this as a two step process? (write out the information to stdout before compilation and then read it from stdout after running) Is there a way to do it in one go, at the right time for ReFrame?